src/RpcServer/RpcServer.Wallet.cs

```diff
@@ -348,20 +347,29 @@ private JObject GetVerificationResult(UInt160 scriptHash, ContractParameter[] ar
             {
                 throw new RpcException(-100, "Unknown contract");
             }
-            var methodName = "verify";
+            var md = contract.Manifest.Abi.GetMethod("verify", -1);
+            if (md is null)
+                throw new RpcException(-101, $"The smart contract {contract.Hash} haven't got verify method.");
+            if (md.ReturnType != ContractParameterType.Boolean)
+                throw new RpcException(-102, "The verify method doesn't return boolean value.");
 
-            Transaction tx = signers == null ? null : new Transaction
+            Transaction tx = new Transaction
             {
-                Signers = signers.GetSigners(),
-                Attributes = Array.Empty<TransactionAttribute>()
+                Signers = signers == null ? new Signer[] { new() { Account = scriptHash } } : signers.GetSigners(),
+                Attributes = Array.Empty<TransactionAttribute>(),
+                Witnesses = signers?.Witnesses,
+                Script = new[] { (byte)OpCode.RET }
             };
-            ContractParametersContext context = new ContractParametersContext(snapshot, tx);
-            wallet.Sign(context);
-            tx.Witnesses = context.Completed ? context.GetWitnesses() : null;
-
             using ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Verification, tx, snapshot.CreateSnapshot(), settings: system.Settings);
-            engine.LoadScript(new ScriptBuilder().EmitDynamicCall(scriptHash, methodName, args).ToArray(), rvcount: 1);
+            engine.LoadContract(contract, md, CallFlags.ReadOnly);
+            if (args.Length > 0)
+            {
+                using ScriptBuilder sb = new ScriptBuilder();
+                for (int i = args.Length - 1; i >= 0; i--)
+                    sb.EmitPush(args[i]);
 
+                engine.LoadScript(new Script(sb.ToArray()), configureState: p => p.CallFlags = CallFlags.None);
+            }
             JObject json = new JObject();
             json["script"] = Convert.ToBase64String(contract.Script);
```

Comment:
And let the resulting `script` be the InvocationScript (which is `sb.ToArray()`). Because 1) the user is aware of the contract being invoked and 2) the user should have the ability to check that all the parameters were converted as expected.

Reply:
See https://github.com/nspcc-dev/neo-go/pull/1825#discussion_r592256453.

Reply:
What `script` mean if `sb` is empty?

Reply:
That no parameters were pushed on the stack before `verify` execution.